core/pkg/net/ssl/ssl.go

```diff
@@ -125,6 +127,56 @@ func AddOrUpdateCertAndKey(name string, cert, key, ca []byte) (*ingress.SSLCert,
 	}, nil
 }
 
+// AddOrUpdateCertAuth creates a .pem file with the specified CAs to be used in Cert Authentication
+func AddOrUpdateCertAuth(name string, ca []byte) (*ingress.SSLCert, error) {
+
+	caName := fmt.Sprintf("ca-%v.pem", name)
```

Comment:
From [here](https://github.com/kubernetes/ingress/pull/235/files#diff-ebe49828d74f82381b8859fe35827d8cR126), the 'name' is the nsSecName, which is the secretName, which is the namespace/secret (with the / replaced by -).

So, each secret might create a different file. In our flow, if it's a CA secret it will be something like ca-namespace-secret.pem. If it's a Keypair secret it's something like namespace-secret.pem.

I think the only thing that would need to be clarified here (or somewhere else, like the docs) is that you need separate secrets for keypair and CA authentication, otherwise it will assume this is a Keypair and will not use this as a CA authentication.